lib/safira/accounts/accounts.ex

```diff
@@ -94,6 +94,20 @@ defmodule Safira.Accounts do
     |> Repo.preload(:badges)
   end
 
+  def get_attendee_by_discord_association_code(discord_association_code) do
+    try do
+      Repo.get_by(Attendee, discord_association_code: discord_association_code)
+    rescue
+      _e in Ecto.Query.CastError -> nil # since this code is sent by an user and it should be an UUID
+    end
+  end
+
+  def get_attendee_by_discord_id(discord_id) do
+      Repo.get_by(Attendee, discord_id: discord_id)
+      |> Repo.preload(:badges)
+      |> Repo.preload(:user)
+  end
```

Comment:
I created this method to be used by other controllers for any request coming from the discord bot. But one other option, which would garantee an higher level of abstraction, would be creating a show method in the DiscordAssociationController which, given an discord_id, returns the attendee's ID. This way, any time the discord bot wanted to make a request to our applicationit would first need to call the show method first, to get the attendee's ID and then make the request with that ID (except for the first association request, which only calls the create method). Although this would double the number of calls to our application, it would enable otther controllers to not need to worry getting the attendee by it's discord ID but rather by its normal ID, as it is already done by most controllers.

What are your opinions on this?

Reply:
If the discord bot is used extensively, it may be worth the extra work on performance. How many cases we need the user ID?

Reply:
If i recall correctly the bot will interact with the backend in order to reddem the companie's badges and when companies want to reward participants for their enthusiasm.